### ISSUE TITLE
Could we add an option to make use of a VCS platform built in archives?

### ISSUE THREAD

**Issue by 7ochem** [composer/satis#347]
I've had this idea when using Satis together with some private repo's on a Bitbucket Server instance. Bitbucket Server has an archive API just like GitHub or Bitbucket.org.

I've been going through Composer's Vcs driver classes to see how Composer is generating dist URL's for packages hosted on GitHub or Bitbucket.org and it seemed to me we could do the very same with Satis.

I would suggest to add an option to the `build` command to indicate to use archive API's where possible, so Satis will generate a `dist` section with a reference to the zip or tar of the package on the platform.

Additionally it would be great to have a Bitbucket Server driver of course (or a possibility to hook in your own drivers).

I am aware that this issue/feature request is a duplicate of issue #196, but this one covers all VCS platforms that have an archive API, not only GitHub.

**Comment by stof** [comment]
Regarding a Bitbucket Server driver, this should be a feature request on composer itself, as this is where VCS drivers are living.

**Comment by CybotTM** [comment]
@stof , i do not agree, IMHO i have this central package repository, this is my first and only responsible authority to fetch any packages in my domain.
Th package it self can come from various sources.
I would configure my Satis to hold any package as zip archive for download - thats fine - but a waste of resources - because most of the origins for packages can provide the zip archive themself.

So it would be nice if i can stay with my central (domain) package repository and don't care about where a originated from.

So i would expect a feature in Satis to utilize the origins API to zip archive and pass them through or redirect the download to them.

**Comment by alcohol** [comment]
@CybotTM your reply makes no sense at all given the context.

**Comment by stof** [comment]
@CybotTM Satis *does* use upstream archives URLs when you don't configure it to generate its own ones. The issue here is that Composer does not have support for using archives from a bitbucket server repository (because it has no special handling for bitbucket server, and so uses the generic git driver, and so cannot have bitbucket-server-specific features like identifying the archive URL).

Satis itself does not implement *any* logic about getting the package info. It just relies on Composer to do that This is why I said it should be a feature request on Composer.

**Comment by CybotTM** [comment]
@alcohol , thanks ... helpful

@stof , yes, i was confused, thanks for clarifying, i had the same discussion internally meanwhile.
Sorry for the noise.
